### DISCUSSION TITLE
Add customized component to a foundry PDK

### DISCUSSION THREAD

**Discussion by sheronlian** [gdsfactory/gdsfactory#2490] [Q&A] [answered]
I self designed a grating coupler and created the gds layout using the grating_coupler_rectangular_arbitrary function and now I'm hoping to incorporate this design in a foundry run following the foundry's DLC. What is the easiest way to do this and where can I find some guidelines on how to start building this component? I would imagine that I would have to map my design to the correct layers provided by the foundry and etc...

**Comment by joamatab** [comment]
We have tutorials for doing this

Does your foundry have a gdsfactory pdk? That would save you time

**Reply by sheronlian** [reply]
Yes, I'm designing using PDK from AIM photonics. So I'm wondering if I should design my customized component first and add the gds file to the component library or if I could just create a function that would create my customized component and just call the function whenever I want to use it. A tutorial would be helpful! Thank you so much.

**Reply by joamatab** [reply, marked as answer]
I hope this helps

https://gdsfactory.github.io/gdsfactory/notebooks/08_pdk_examples.html
